src/PROPOSAL/PROPOSAL/crosssection/CrossSection.h

```diff
@@ -228,9 +229,20 @@ class CrossSection : public CrossSectionBase {
     std::unique_ptr<std::vector<std::tuple<double, de2dx_ptr>>> de2dx;
 
     double lower_energy_lim;
+    double min_stochastic_energy;
     InteractionType interaction_type;
     std::string param_name;
 
+private:
+    template <typename T> static double CalculateMinStochasticEnergy(T const& m)
+    {
+        double min_energy = std::numeric_limits<double>::infinity();
+        for (auto const& i : m)
```

Comment:
This could just be std min_element

Reply:
```
    return std::min_element(
        m.cbegin(), m.cend(),
        [](const auto& kv1, const auto& kv2){return kv1.second.energy <= kv2.second.energy;}
    )->second.energy;
```

Reply:
But this does not exist in c++14. It is a `c++17` feature :see_no_evil:

Reply:
Ah, sorry, thought PROPOSAL was already C++17

Reply:
Done in 8b25d6febb5752eb552d19384c9e65bceade5f24.
But I am not sure if it's reduces the code complexity.